test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionExecutorTests.cs

```diff
@@ -99,17 +92,16 @@ public class ControllerActionExecutorTests
             var methodWithTaskOfTaskOfIntReturnType = new MethodWithTaskOfTaskOfIntReturnType(_controller.TaskOfTaskAction);
 
             // Act
-            var result = await (Task<int>)(await ControllerActionExecutor.ExecuteAsync(
-                                                                        methodWithTaskOfTaskOfIntReturnType.GetMethodInfo(),
-                                                                        _controller,
-                                                                        actionParameters));
+            var result = await (Task<int>)( await ExecuteAction(usingExecutor, methodWithTaskOfTaskOfIntReturnType, _controller, actionParameters));            
 
             // Assert
             Assert.Equal(inputParam1, result);
         }
 
-        [Fact]
-        public async Task AsyncAction_WithAsyncKeywordThrows()
+        [Theory,
+         InlineData(true),
+         InlineData(false)]
```

Comment:
Instead of doing this to the tests, can we just remove the `MethodInfo` support?